addons/repair/models/repair.py

```diff
@@ -111,11 +111,23 @@ def _default_picking_type_id(self):
         'procurement.group', 'Procurement Group',
         copy=False)
     location_id = fields.Many2one(
-        'stock.location', 'Location',
+        'stock.location', 'Component Source Location',
         compute="_compute_location_id",
         store=True, readonly=False, required=True, precompute=True,
         index=True, check_company=True,
+        help="This is the location where the components of product to repair is located.")
+    default_product_location_src_id = fields.Many2one(
+        'stock.location', 'Product Source Location',
+        compute="_compute_default_product_location_src_id",
```

Comment:
make it related to picking type + rename `product_location_src_id`

same for "dest" location

Reply:
Hello @Whenrow,

We cannot set this field as a related field because whenever the product's location changes whether from the RO, a return, or another source it resets to the default value. That’s why we need to make it computed, allowing us to modify the value as needed.

As per this review by MGM! 
(what is happening here? Always use the default locations of the operation type. demo: ﻿[﻿https://drive.google.com/file/d/1p8KvmXqW97OngWoKcO_8zSDPgxMueouE/view?usp=drivesdk](https://drive.google.com/file/d/1p8KvmXqW97OngWoKcO_8zSDPgxMueouE/view?usp=drivesdk) ) 

Thank you!

Reply:
ok